java.lang.NullPointerException  openfoodfacts.github.scrachx.openfood.utils.LocaleHelper.getLocale

```  
Caused by: java.lang.NullPointerException: 
  at openfoodfacts.github.scrachx.openfood.utils.LocaleHelper.getLocale (LocaleHelper.java:145)
  at openfoodfacts.github.scrachx.openfood.utils.LocaleHelper.getLocale (LocaleHelper.java:125)
  at openfoodfacts.github.scrachx.openfood.views.ContinuousScanActivity.attachBaseContext (ContinuousScanActivity.java:679)
```

Perhaps already fixed ?

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/HomeFragment.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/HomeFragment.java
@@ -55,6 +55,7 @@ public class HomeFragment extends NavigationBaseFragment implements CustomTabAct
     private OpenFoodAPIService apiClient;
     private SharedPreferences sp;
     private String taglineURL;
+    private Disposable disposable;
 
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
@@ -118,11 +119,13 @@ public class HomeFragment extends NavigationBaseFragment implements CustomTabAct
                             .putString("pass", "")
                             .apply();
 
-                        new MaterialDialog.Builder(getActivity())
-                            .title(R.string.alert_dialog_warning_title)
-                            .content(R.string.alert_dialog_warning_msg_user)
-                            .positiveText(R.string.txtOk)
-                            .show();
+                        if(getActivity()!=null) {
+                            new MaterialDialog.Builder(getActivity())
+                                .title(R.string.alert_dialog_warning_title)
+                                .content(R.string.alert_dialog_warning_msg_user)
+                                .positiveText(R.string.txtOk)
+                                .show();
+                        }
                     }
                 }
 
@@ -131,6 +134,15 @@ public class HomeFragment extends NavigationBaseFragment implements CustomTabAct
                     Log.e(HomeFragment.class.getName(), "Unable to Sign-in");
                 }
             });
+        }
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+        //stop the call to off to get total product counts:
+        if (disposable != null && !disposable.isDisposed()) {
+            disposable.dispose();
         }
     }
 
@@ -145,26 +157,33 @@ public class HomeFragment extends NavigationBaseFragment implements CustomTabAct
             .subscribe(new SingleObserver<Search>() {
                 @Override
                 public void onSubscribe(Disposable d) {
-                    updateTextHome(productCount);
+                    disposable=d;
+                    if(isAdded()) {
+                        updateTextHome(productCount);
+                    }
                 }
 
                 @Override
                 public void onSuccess(Search search) {
-                    int totalProductCount = productCount;
-                    try {
-                        totalProductCount = Integer.parseInt(search.getCount());
-                    } catch (NumberFormatException e) {
-                        Log.w(HomeFragment.class.getSimpleName(), "can parse " + search.getCount() + " as int", e);
+                    if(isAdded()) {
+                        int totalProductCount = productCount;
+                        try {
+                            totalProductCount = Integer.parseInt(search.getCount());
+                        } catch (NumberFormatException e) {
+                            Log.w(HomeFragment.class.getSimpleName(), "can parse " + search.getCount() + " as int", e);
+                        }
+                        updateTextHome(totalProductCount);
+                        SharedPreferences.Editor editor = sp.edit();
+                        editor.putInt("productCount", totalProductCount);
+                        editor.apply();
                     }
-                    updateTextHome(totalProductCount);
-                    SharedPreferences.Editor editor = sp.edit();
-                    editor.putInt("productCount", totalProductCount);
-                    editor.apply();
                 }
 
                 @Override
                 public void onError(Throwable e) {
-                    updateTextHome(productCount);
+                    if(isAdded()) {
+                        updateTextHome(productCount);
+                    }
                 }
             });
 
@@ -179,14 +198,15 @@ public class HomeFragment extends NavigationBaseFragment implements CustomTabAct
     }
 
     private void updateTextHome(int totalProductCount) {
-        textHome.setText(R.string.txtHome);
-        if (totalProductCount != 0) {
-            String txtHomeOnline = getResources().getString(R.string.txtHomeOnline);
-            try {
+        try {
+            textHome.setText(R.string.txtHome);
+            if (totalProductCount != 0) {
+                String txtHomeOnline = getResources().getString(R.string.txtHomeOnline);
+
                 textHome.setText(String.format(txtHomeOnline, totalProductCount));
-            } catch (Exception e) {
-                Log.w(HomeFragment.class.getSimpleName(), "can format text for home", e);
             }
+        } catch (Exception e) {
+            Log.w(HomeFragment.class.getSimpleName(), "can format text for home", e);
         }
     }
 

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/LocaleHelper.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/LocaleHelper.java
@@ -142,9 +142,16 @@ public class LocaleHelper {
     }
 
     public static Locale getLocale(Context context) {
-        Resources resources = context.getResources();
-        Configuration configuration = resources.getConfiguration();
-        final Locale locale = configuration.locale;
+        Locale locale = null;
+        if (context != null) {
+            Resources resources = context.getResources();
+            if (resources != null) {
+                Configuration configuration = resources.getConfiguration();
+                if (configuration != null) {
+                    locale = configuration.locale;
+                }
+            }
+        }
         return locale == null ? Locale.getDefault() : locale;
     }
 
@@ -195,6 +202,9 @@ public class LocaleHelper {
      * @return Locale from locale string
      */
     public static Locale getLocale(String locale) {
+        if(locale==null){
+            return Locale.getDefault();
+        }
         String[] localeParts = locale.split("-");
         String language = localeParts[0];
         String country = localeParts.length == 2 ? localeParts[1] : "";

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/listeners/CommonBottomListener.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/listeners/CommonBottomListener.java
@@ -37,8 +37,8 @@ public class CommonBottomListener implements BottomNavigationView.OnNavigationIt
                 }
                 if (Utils.isHardwareCameraInstalled(context)) {
                     if (ContextCompat.checkSelfPermission(context, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
-                        if (ActivityCompat.shouldShowRequestPermissionRationale(activity, Manifest.permission.CAMERA)) {
-                            new MaterialDialog.Builder(context)
+                        if (activity.hasWindowFocus() && ActivityCompat.shouldShowRequestPermissionRationale(activity, Manifest.permission.CAMERA)) {
+                            new MaterialDialog.Builder(activity)
                                 .title(R.string.action_about)
                                 .content(R.string.permission_camera)
                                 .neutralText(R.string.txtOk)

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ProductAttributeDetailsFragment.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ProductAttributeDetailsFragment.java
@@ -130,7 +130,8 @@ public class ProductAttributeDetailsFragment extends BottomSheetDialogFragment i
     }
 
     private void updateContent(View view, AdditiveName additive) {
-        View exposureEvalTable = view.findViewById(R.id.exposureEvalTable);
+
+
         mpInfantsImage = view.findViewById(R.id.mpInfants);
         mpToddlersImage = view.findViewById(R.id.mpToddlers);
         mpChildrenImage = view.findViewById(R.id.mpChildren);
@@ -144,10 +145,10 @@ public class ProductAttributeDetailsFragment extends BottomSheetDialogFragment i
         spAdultsImage = view.findViewById(R.id.spAdults);
         spElderlyImage = view.findViewById(R.id.spElderly);
 
-        TextView efsaWarning = view.findViewById(R.id.efsaWarning);
-
-        String overexposureRisk = additive.getOverexposureRisk();
-        if (additive.hasOverexposureData()) {
+        if (additive!=null && additive.hasOverexposureData()) {
+            View exposureEvalTable = view.findViewById(R.id.exposureEvalTable);
+            TextView efsaWarning = view.findViewById(R.id.efsaWarning);
+            String overexposureRisk = additive.getOverexposureRisk();
             boolean isHighRisk = "high".equalsIgnoreCase(overexposureRisk);
             if (isHighRisk) {
                 bottomSheetTitleIcon.setImageResource(R.drawable.ic_additive_high_risk);

